debezium-core/src/main/java/io/debezium/relational/TableSchema.java

```diff
@@ -126,7 +127,7 @@ public Envelope getEnvelopeSchema() {
      */
     public Struct keyFromColumnData(Object[] columnData) {
         if (LOGGER.isTraceEnabled()) {
```

Comment:
Any conditional checks of log level are redundant. You can use a lambda (e.g., directly or with [addArgument](https://slf4j.org/manual.html#fluent)) or you can wrap columnData in an encapsulating object to define a custom toString method. The latter is perhaps a refactor for another PR fyi @jpechane

Reply:
The conditional checks and the other toString methods have existed before the change introduced in the PR. I would prefer to keep refactoring of those separate to avoid introducing any new regressions tbh.

Reply:
I would say those changes don't necessarily need to be in a different PR, but they certainly could be in a separate commit from the "reverting" of the performance regression.

For this use case specifically, what about taking the most simple path introducing a method in the `Loggings` class that takes an `Object[]` that internally uses `Arrays.toString` on the argument if TRACE is enabled. It would continue to have negligible overhead and avoids the redundant check.